src/typechecker.cpp

```diff
@@ -7,20 +7,23 @@
 namespace TypeChecker {
 
 TypeChecker::TypeChecker() {
-	m_core.new_type_function(-1); // 0 | function
-	m_core.new_type_function(0);  // 1 | int
-	m_core.new_type_function(0);  // 2 | float
-	m_core.new_type_function(0);  // 3 | string
-	m_core.new_type_function(1);  // 4 | array
-	m_core.new_type_function(1);  // 5 | dictionary
-	m_core.new_type_function(0);  // 6 | boolean
-	m_core.new_type_function(0);  // 7 | unit
-
-	m_core.new_term(1, {}, "builtin int"); // 0 | int(<>)
-	m_core.new_term(2, {}, "builtin float"); // 1 | float(<>)
+	m_core.new_type_function(-1); // 0  | function
+	m_core.new_type_function(0);  // 1  | int
+	m_core.new_type_function(0);  // 2  | float
+	m_core.new_type_function(0);  // 3  | string
+	m_core.new_type_function(1);  // 4  | array
+	m_core.new_type_function(1);  // 5  | dictionary
+	m_core.new_type_function(0);  // 6  | boolean
+	m_core.new_type_function(0);  // 7  | unit
+	m_core.new_type_function(-1); // 8  | sum
+	m_core.new_type_function(-1); // 9  | product
+	m_core.new_type_function(-1); // 10 | record
```

Comment:
This doesn't sound right to me.

A sum (or product, etc) type is a form of type function, it's not a particular type function.

I think `sum`, `product`, and `record` should be values in the `type_function_type` enum

Reply:
To sum up: I think a type function should either be a var, a builtin, a sum, a product, or a record

Reply:
Yea perhaps this is a better way to do it